### ISSUE TITLE
Writing numbers bug

### ISSUE THREAD

**Issue by Roman3349** [minetest/minetest#3170]
When I want write the digit in the dialog for connect to the server or create server, so I can not write digit 1, 2, 4, 5, 6, 7 and 8 on the numeric keypad and digit 4 and 7 above the QWERTY keys.

OS: Linux Mint 17.2 x86_64
Minetest: 0.4.13
Language: czech

Sorry for my English.

**Comment by Wayward1** [comment]
I experience something similar with mine, but only with the numeric keypad. With number lock on, I can only type 9 and 3, while the numbers along the top work fine. I have the same OS and the latest Git.

**Comment by dsohler** [comment]
I can confirm this bug in 0.4.13

~~~~
$ uname -rms
Linux 4.1.6-1-ARCH x86_64

$ pacman -Qi irrlicht | grep Version
Version              : 1.8.2-2
~~~~

Occurred in 0.4.12, too.

**Comment by est31** [comment]
Can somebody compile the user interface example (download irrlicht source cd examples then cd to the user interface dir, then make, then you find the executable in ../../bin/Linux/) of irrlicht, and check whether it occurs there too? This smells of irrlicht's horrible input handling. Also, wondering whether it is fixed in 1.9.

**Comment by kilbith** [comment]
Related : https://github.com/minetest/minetest/issues/302

**Comment by ShadowNinja** [comment]
I'm quite sure this is an Irrlicht bug.  I've heard of this before, although it was just the `7` key last time.

**Comment by RobertZenz** [comment]
#1898

**Comment by paramat** [comment]
Now fixed in irrlicht 1.9?

**Comment by est31** [comment]
Even if it gets fixed in irrlicht 1.9 we will be stuck with it until a fix gets added to the 1.8 branch as well, as most of minetest users still use irrlicht 1.8 (and it will stay that way until an irrlicht 1.9 release comes out).

**Comment by deltaflyer4747** [comment]
Just wanted to say that i just tried to compile against an 1.9 irrlicht; compilation goes smoothly, just some minor warnings, however when i try to launch the game it fails with:

2017-11-19 09:19:33: WARNING[Main]: Couldn't find a locale directory!
X Error of failed request: BadWindow (invalid Window parameter)
Major opcode of failed request: 3 (X_GetWindowAttributes)
Resource id in failed request: 0x1
Serial number of failed request: 26
Current serial number in output stream: 27

**Comment by deltaflyer4747** [comment]
a) same thing as in #845

b) Just wanted to say that i just tried to compile against an 1.9 irrlicht; compilation goes smoothly, just some minor warnings, however when i try to launch the game it fails with:

2017-11-19 09:19:33: WARNING[Main]: Couldn't find a locale directory!
X Error of failed request: BadWindow (invalid Window parameter)
Major opcode of failed request: 3 (X_GetWindowAttributes)
Resource id in failed request: 0x1
Serial number of failed request: 26
Current serial number in output stream: 27

**Comment by paramat** [comment]
Duplicated post @deltaflyer4747 shall i delete the first one?

**Comment by deltaflyer4747** [comment]
please delete the one in 845, thanks (and if possible, this one with yours as well)

**Comment by paramat** [comment]
We'll keep #845 closed and discuss here.
Also deleted your duplicated post above.

**Comment by sfan5** [comment]
duplicate of now closed #302
